app/waf/src/controllers/adc.controller.ts

```diff
@@ -250,16 +255,27 @@ export class AdcController {
 
       case 'setup':
         let mgr = await BigIpManager.instanlize({
-          username: 'admin',
-          password: 'admin',
-          ipAddr: 'localhost', //adc.management.ipAddress,
-          port: 8443, //adc.management.tcpPort,
+          username: adc.management.username,
+          password: adc.management.password,
+          ipAddr: adc.management.ipAddress,
+          port: adc.management.tcpPort,
         });
         let ready = await mgr.checkAndWaitBigipReady(5 * 1000);
         if (ready) {
-          this.logger.debug('start to do onbarding here.');
-          // DO here.
-          return {id: adc.id};
+          try {
+            this.logger.debug('start to do onbarding');
+            let doMgr = await OnboardingManager.instanlize();
+            let doBody = await doMgr.assembleDo(adc);
+            this.logger.debug(
+              'Json used for onboarding: ' + JSON.stringify(doBody),
+            );
+            let doRes = await doMgr.onboarding(doBody);
```

Comment:
How long will it take? Will we plan not to wait here in the future?

Reply:
Generally, it can be over in < 20s. I added a 'TODO' in the code to think it in the future whether to async execute it.

Reply:
OK